src/lib/autoinstall/widgets/editable_combo_box.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Autoinstallation
+  module Widgets
+    # Mixin to extend the #value= method for editable ComboBox widgets
+    #
+    # Ideally, this behaviour might be implemented in the original
+    # CWM ComboBox class.
+    #
+    # @example Define an editable combo box
+    #  class MyComboBox < CWM::ComboBox
+    #    include EditableComboBox
+    #
+    #    def opt
+    #      [:editable]
+    #    end
+    #  end
+    #
+    # As you can see in the example, you need to define the
+    # `#opt` method to include the `:editable` option.
+    module EditableComboBox
+      # Changes the list of items
+      #
+      # @macro seeComboBox
+      def change_items(new_items)
+        @items = new_items
```

Comment:
when you do this, you should also redefine `def items` and return this instance variable, otherwise it won't survive redraw.
In general I am not sure if mixin is good idea here. I would create new class that redefine the needed items. So usage would be simply `< CWM::EditableComboBox`. Of course another option is to read opt in CWM class itself and if it contain `:editable` then use this new behavior in its base class. I am not sure what is better option, but probably separate class makes more sense for me.

Reply:
I am usually not sure whether it is better to use inheritance or a mixin. If I wanted to have another additional behavior, I might need to add another level to the inheritance chain.

On the other hand, in this case I think that, as you have proposed, we should implement this behavior when `editable` is included in the widget options so this mixin will not be needed at all. I will open a PR.